### PR TITLE
Promote Python SDK in sdks/python-sdk from POC to a complete and supported SDK

### PR DESCRIPTION
**Pull request by ibrahimsel**
As mentioned in #824:

This PR aims to promote the existing Python SDK under `sdks/python-sdk` from a proof-of-concept to a structured, maintainable, and supported SDK.

### Summary of changes

This PR introduces the following improvements:

* Converts the SDK to a proper Python package with a `pyproject.toml`, explicit dependency definitions, and reproducible builds.
* Establishes a clear SDK structure with separated concerns for:

  * Core client logic
  * Typed models
  * Provider and extension interfaces
  * Adds basic test coverage to validate core functionality
  * Includes minimal but functional examples to demonstrate intended usage
  * Improves documentation to clarify installation, usage, and extension points

### Non-goals

This PR intentionally does **not** attempt to:

* Implement every possible Symphony API or feature.
* Lock the SDK API surface as final or frozen.
* Optimize for advanced performance or edge cases.

Those concerns are deferred until the SDK is adopted and exercised by users.

### Potential Follow-ups

* Expanding API coverage as Symphony APIs stabilize.
* Strengthening test coverage and possibly CI integration.
* Publishing versioned releases and formal support guarantees.
* Fix `ruff` and `mypy` warnings/errors

---

### Notes For Reviewers
- I intentionally did not add any license to the sdk's root or to `pyproject.toml` as I couldn't quite choose if we use EPL 2.0 or MIT for Eclipse Symphony. (You'll see license headers in source code, I didn't remove them since they're coming from Muto Agent's repository). If someone enlightens me of the license requirements, I could update accordingly
- I did not touch the existing POC `python-sdk`. It is still there
- You could manually try the SDK quickly before merge by following the guidelines in [Examples](https://github.com/ibrahimsel/symphony/blob/symphony-python/sdks/symphony-python/examples/README.md)
- Unit test coverage is currently 76%
Screenshot from 2026-01-13 17-05-02

- After the SDK is good to merge, please publish it to [PyPI](https://pypi.org/). We need this SDK in [Muto Agent](https://github.com/eclipse-muto/agent). Since the package is not published yet, we're currently keeping a tiny version of this SDK in `Muto Agent`'s repo. If PyPI package gets published, we'll migrate to that. @Haishi2016

### CODE REVIEW

File: sdks/symphony-python/src/symphony_sdk/api_client.py [hunk 1]
```diff
@@ -0,0 +1,652 @@
...
+        access_token = data.get("accessToken")
+        if not access_token:
+            raise SymphonyAPIError("No access token in authentication response")
+
+        self._access_token = access_token
+        # Assume token is valid for ~1 hour
+        # Use timezone-aware UTC datetimes and truncate seconds/micros
+        self._token_expiry = datetime.now(timezone.utc).replace(microsecond=0, second=0)
+        self._token_expiry = self._token_expiry.replace(
+            minute=(self._token_expiry.minute + 50) % 60
```
Comment (Haishi2016) [inline sdks/symphony-python/src/symphony_sdk/api_client.py]: This logic doesn't add 50 minutes; for example if minute is 10 then this becomes 0. Maybe: datetime.now(timezone.utc) + timedelta(minutes=50)

**Review by Haishi2016 — COMMENTED** [review]
Great PR. Some comments are added.

File: sdks/symphony-python/src/symphony_sdk/models.py [hunk 2]
```diff
@@ -0,0 +1,670 @@
...
+
+
+@dataclass
+class ComponentSpec:
+    name: str = ""
+    type: str = ""
+    routes: List[RouteSpec] = None
+    constraints: str = ""
+    properties: Dict[str, str] = None
+    depedencies: List[str] = None
```
[the remark this answers is not in the document]
Comment (ibrahimsel) [reply sdks/symphony-python/src/symphony_sdk/models.py]: Fixed the typo in SDK in commit 0b7c18159e87672c891e893462f9ff85bee7af26 with a note for you:

took this part copy-paste from [`sdk_poc.py`](https://github.com/eclipse-symphony/symphony/blob/be7d32993023c1d61d230b5ed366ef9b39965aa9/sdks/python-sdk/sdk_poc.py#L85) without noticing. Then I noticed there are multiple occurences of `depedencies` typo in the codebase. Screenshot below. 

image

File: sdks/symphony-python/src/symphony_sdk/api_client.py [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+
+    def health_check(self) -> bool:
+        """Perform a basic health check of the Symphony API.
+
+        Returns:
+            True if API is accessible, False otherwise
+        """
+        try:
+            # Try a simple endpoint that doesn't require auth
+            response = self._make_request("GET", "/greetings", json={"foo": "bar"})
```
[the remark this answers is not in the document]
Comment (ibrahimsel) [reply sdks/symphony-python/src/symphony_sdk/api_client.py]: fixed in commit 858e954d94fb813ede11f5f3c9c0f262abcd684e

File: sdks/symphony-python/src/symphony_sdk/api_client.py [hunk 4]
```diff
@@ -0,0 +1,652 @@
+#
+# Copyright (c) 2025 Composiv.ai
```
[the remark this answers is not in the document]
Comment (ibrahimsel) [reply sdks/symphony-python/src/symphony_sdk/api_client.py]: Resolved in commit c71dff1c51c6e96f5645afae2a289a5bdc9d881e

File: sdks/symphony-python/src/symphony_sdk/api_client.py [hunk 5]
```diff
@@ -0,0 +1,652 @@
...
+        self._ensure_authenticated()
+
+        response = self._make_request(
+            "GET",
+            f"/targets/registry/{target_name}",
+        )
+
+        return self._handle_response(response)
+
+    def list_targets(self) -> List[Dict[str, Any]]:
```
[the remark this answers is not in the document]
Comment (Haishi2016) [reply sdks/symphony-python/src/symphony_sdk/api_client.py]: I see. We can keep the current approach as it is. Or, if we want to make it a bit more clearer, maybe use Any? def _handle_response(...) -> Any. Or, have a JSONType: JSONType = Union[Dict[str, Any], List[Dict[str, Any]]]
def _handle_response(...) -> JSONType. And because the method may return None, make it Optional[JsonType]? It's your call.

[the remark this answers is not in the document]
Comment (ibrahimsel) [reply sdks/symphony-python/src/symphony_sdk/api_client.py]: > Or, have a JSONType: JSONType = Union[Dict[str, Any], List[Dict[str, Any]]]
> def _handle_response(...) -> JSONType. And because the method may return None, make it Optional[JsonType]?

I'll try this soon

[the remark this answers is not in the document]
Comment (ibrahimsel) [reply sdks/symphony-python/src/symphony_sdk/api_client.py]: implemented your suggestion for `handle_response` in 90a9cc500a5d1232d93c5537a5494aed8516e02d

Left the return types of `list_*` methods as is. Because callers would need to narrow the type before using it as a list, which would be unnecessary friction in my opinion

**Review by Haishi2016 — APPROVED** [review]
